### PR TITLE
Migrating repositories

### PR DESCRIPTION
**Pull request by janstaelensskyline**

### CODE REVIEW

**Comment by janstaelensskyline** [comment]
Note concerning Reviewing. The old CICD documents concerning gerrit & jenkins, now located under CI/CD>Skyline Communications>Gerrit and Jenkins  did not have their contents changed but were simply moved.

File: develop/CICD/GitHub/GitHub_Starter_Workflows.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+In the free version of GitHub, you will not see starter workflows in the UI directly. In this case, perform the following steps:
+
+1. Navigate to the *workflow-templates* folder in the *.github* repository of your organization.
+
+   For members of Skyline Communications, this is <https://github.com/SkylineCommunications/.github/tree/main/workflow-templates>.
+
+1. Select the workflow template that matches the content of your repository and look at the source code.
+
+1. Copy that and navigate to your own Repository Action tab.
```
Comment (MariekeGO) [inline develop/CICD/GitHub/GitHub_Starter_Workflows.md:37]: Is "Repository Action" the exact name of the tab? Or is it "Actions" as mentioned above?

File: develop/CICD/GitHub/GitHub_Starter_Workflows.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+At present, GitHub allows community starter workflows to be placed only in the .github directory of an organization, rather than in the global overview. To help your organization obtain new workflows from Skyline Communications even though these cannot be placed in the global overview, we have created a [public repository](https://github.com/SkylineCommunications/Skyline-Starter-Workflows) that you can [fork](https://github.com/SkylineCommunications/Skyline-Starter-Workflows/fork) into your own organization.
+
+Once you have [set up that fork](#setting-up-organization-starter-workflows), you will automatically receive pull requests on your organization's .GitHub repository with changes to the starter workflows. Users with the appropriate authentication will be able to accept and merge these pull requests to benefit from any new or modified workflow files.
+
+This allows your organization to easily adjust or add workflows to meet your specific needs and to handle any necessary merging through pull requests.
+
+## Setting up organization starter workflows
+
+1. Optionally, start by creating a GitHub user specifically for bot activities, e.g. name:CICDOrganization.
```
Comment (MariekeGO) [inline develop/CICD/GitHub/GitHub_Starter_Workflows.md:59]:
```suggestion
1. Optionally, start by creating a GitHub user specifically for bot activities, e.g. with the name *CICDOrganization*.
```

File: develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+---
+uid: github_reusable_workflows_nuget_solution_master_workflow
+---
+
+# NuGet solution master workflow
+
+> [!IMPORTANT]
+> This workflow is intended for internal use at Skyline Communications as it uses Skyline-specific validation. However, you can use it as a baseline and adjust it for use elsewhere by replacing the validation step with your own.
+
+The NuGet solution master workflow is designed to run on repositories containing the [DataMiner NuGet Package Solution](xref:Automation_scripts_as_a_Visual_Studio_solution) provided by the DIS extension in Visual Studio or from [Skyline.DataMiner.VisualStudioTemplates](https://www.nuget.org/packages/Skyline.DataMiner.VisualStudioTemplates#readme-body-tab).
```
Comment (MariekeGO) [inline develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md:10]: Is this first cross-reference correct? I don't find any mention of the DataMiner NuGet Package Solution on the target page.

File: develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+- Skyline quality gate:
+
+  - [Validate NuGet metadata](#validate-nuget-metadata)
+  - [Building](#building)
+  - [Unit tests](#unit-tests)
+  - [Analyze](#analyze)
+  - [Quality gate](#quality-gate)
+
+  If the library passes these checks, the job will archive and provide the created NuGet package as an artifact, which the next job will attempt to sign.
```
Comment (MariekeGO) [inline develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md:26]: What is it that is archived: the job or the NuGet package? Could you rephrase this sentence so no confusion is possible?

File: develop/CICD/Skyline Communications/Github/Migration_From_Gerrit_To_Github.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  
+   > [!NOTE]
+   > Optionally, you can choose to continue to work with your existing local repository (in your file system). You can do so by removing the original origin location with the following commands:
+   >
+   > ```bash
+   > git remote remove origin
+   > git remote rename origin2 origin
+   > ```
+
+1. Consider the [validation requirements](xref:github_validation_requirements) and add a README.md file and remove the JenkinsFile when you start working on your branch.
```
Comment (MariekeGO) [inline develop/CICD/Skyline Communications/Github/Migration_From_Gerrit_To_Github.md:55]: What do you mean with "Consider the validation requirements"? What should the user actually do? Make sure these requirements are met? Also, what's the connection to the second part of the step, after "and"? Or can the second part become a step of its own in the procedure?

**Review by MariekeGO — CHANGES_REQUESTED** [review]

File: develop/CICD/GitHub/GitHub_Starter_Workflows.md (hunk 1 above)
Comment (janstaelensskyline) [reply develop/CICD/GitHub/GitHub_Starter_Workflows.md:37]: "Actions"

File: develop/CICD/Skyline Communications/Github/Migration_From_Gerrit_To_Github.md (hunk 5 above)
Comment (janstaelensskyline) [reply develop/CICD/Skyline Communications/Github/Migration_From_Gerrit_To_Github.md:55]: The user should make sure he meets the validation requirements to avoid getting failures when running our workflows.
The README file did not exist on gerrit, so must be added when migrating.
The JenkinsFile is no longer usefull on github so should be removed.
These items however, cannot be done in 'bulk' like the migration described earlier. They can only be tackled ad-hoc whenever someone starts working on a specific branch.

File: develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md (hunk 4 above)
Comment (janstaelensskyline) [reply develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md:26]: The NuGet package is archived, which means it becomes available as an Artifact in the workflow.
Perhaps:

If the library passes these checks, the job will archive the created NuGet package. This package is then available as an artifact on the workflow, which a user or the next jobs can access. The next job will attempt to sign this package.

File: develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+# GitHub reusable workflows
+
+Where a [GitHub Action](xref:Deploying_Automation_scripts_from_a_GitHub_repository) can provide an easy access point to a single key action, the reusable workflow takes this up a notch.
+A GitHub reusable workflow allows the re-use of a combination of many different GitHub actions and other scripts, running across several jobs combined into a single easy call.
+
+It provides versioning and can be called in such a way to always use the latest stable version.
+This contrary to [Starter workflows](xref:github_starter_workflows) that copy and paste a template workflow without the ability to run the 'latest version'.
+
+Skyline Communications provides most quality control and automation through reusable workflows, allowing us to improve and tweak them based on user feedback or corporate changes and immediately apply the changes to any workflow using them on GitHub.
```
[the remark this answers is not in the document]
Comment (janstaelensskyline) [reply develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows.md:11]: The reusable workflow code runs under version-control and understands the concept of 'latest release'. This allows other workflows to easily request the latest stable version every run, without needing to make changes to the workflow.

This is contrary to [Starter workflows](xref:github_starter_workflows) that involve a simple copy-paste of the workflow code at one point in time (similar to taking a snapshot). Any changes we do to the code of a start-workflow will not automatically trickle down.

File: develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+These reusable workflows are public and can be accessed by any organization. They can also be forked and used as a starting point for other CI/CD engineers, who can adjust them to match their organization's quality standards.
+
+> [!NOTE]
+> If you are a CI/CD engineer looking to save time when building workflows, consider starting from the source code of a reusable workflow. Do note that reusable workflows are internally complex, often combining multiple tools across several jobs and steps to meet quality standards, automation requirements, and edge cases. Many out-of-the-box actions provided online may be lacking in functionality, requiring additional code to prepare a solution before these standard actions are used. We prefer using dotnet tools over scripting code because they allow for clean versioning, reusability, and unit testing of our code. The source code for these tools is currently internal to Skyline Communications, but the dotnet tools themselves are available for use and can be found here:
```
[the remark this answers is not in the document]
Comment (janstaelensskyline) [reply develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows.md:16]: Sorry must have gotten distracted and forgot to add: 
https://www.nuget.org/packages?q=Skyline.Communications.CICD.Tools&frameworks=&tfms=&packagetype=&prerel=true&sortby=relevance

File: develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (MariekeGO) [reply develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows.md:11]: Can I adjust this sentence like this?
```suggestion
Skyline Communications provides most quality control and automation through reusable workflows. Any changes we implement when we improve or tweak these workflows, for example based on user feedback, will also immediately be applied to any workflows on GitHub based on them.
```

**Review by MariekeGO — APPROVED** [review]
